.github/workflows/code-checker.yml

```diff
@@ -0,0 +1,47 @@
+name: Code Checker
```

Comment:
we should change the name to `PR workflow` It makes more sense because later we are going to add many more jobs in this.

Please change the workflow file name to `pull-request.yml`. This will make more sense.